spec/u3d/unity_project_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'u3d/installer'
+require 'u3d_core/helper'
+require 'yaml'
+
+describe U3d do
+  describe U3d::UnityProject do
+    describe '#editor_version' do
+      before(:all) do
+        @config = { 'm_EditorVersion' => 'test' }
+        @project = U3d::UnityProject.new('foo')
+      end
+
+      before(:each) do
+        allow(File).to receive(:read) { 'bar' }
+        allow(YAML).to receive(:load) { @config }
+      end
+
+      describe 'when under Linux' do
+        before(:each) do
```

Comment:
I was originally doing so, but the doubles and allows are not supported in before(:all), and throw 

> The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported.

[More on that](https://relishapp.com/rspec/rspec-mocks/docs/basics/scope)